sdl-compliance-pipeline.yml

```diff
@@ -0,0 +1,49 @@
+trigger:
+  batch: 'true'
+  branches:
+    include:
+    - main
+  paths:
+    exclude:
+    - '*README.md'
+    - 'docs/*'
+
+pool:
+  vmImage: 'windows-2022'
+
+variables:
+  LGTM.UploadSnapshot: true
+  solution: '**/*.sln'
+  msbuildPath: '"%ProgramFiles(x86)%\Microsoft Visual Studio\2022\Enterprise\MSBuild\Current\Bin\MSBuild.exe"'
+  BuildPlatform: x64
+  BuildConfiguration: Release
+
+jobs:
+  - job: SDLCompliance
+    displayName: 'Running SDL Compliance Policy checks'
+    continueOnError: true
+    steps:
+    - task: Semmle@1
+      displayName: 'Run CodeQL (Semmle)'
+      env:
+        SYSTEM_ACCESSTOKEN: $(System.AccessToken)
+      inputs:
+        sourceCodeDirectory: '$(Build.SourcesDirectory)\LogMonitor'
+        language: 'cpp'
+        buildCommands: '$(msbuildPath) $(BUILD.SourcesDirectory)\$(solution) /t:clean #
+                        $(msbuildPath) $(BUILD.SourcesDirectory)\$(solution) /p:platform=$(BuildPlatform) /p:configuration=$(BuildConfiguration)'
+        querySuite: 'Recommended'
+        timeout: '1800'
+        ram: '16384'
```

Comment:
is this 16GB? Why so much ram?

Reply:
Part of those performance initiatives around LogMonitor could include investigating the minimum system resources to build Log Monitor.

Reply:
From the sample CodeQL [codes](https://eng.ms/docs/cloud-ai-platform/devdiv/one-engineering-system-1es/1es-docs/codeql/codeql-for-yaml-based-azure-pipelines#:~:text=ram%3A-,16384,-Required) 16384 is the value provided maybe if it worth exploring if we want to half this to save on resources but I do not see any warnings against having this.

Reply:
Please see this doc: [Recommended hardware resources for running CodeQL](https://docs.github.com/en/enterprise-cloud@latest/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/recommended-hardware-resources-for-running-codeql)

Reply:
Updated to use 8GB